hudi-common/src/main/java/org/apache/hudi/metadata/BaseFileRecordParsingUtils.java

```diff
@@ -0,0 +1,125 @@
+   * If a base file is added to an existing file group, we read previous base file in addition to the latest base file of interest. Find deleted records and generate RLI Metadata records
+   * for the same in addition to new insert records.
+   * @param basePath base path of the table.
+   * @param writeStat {@link HoodieWriteStat} of interest.
+   * @param writesFileIdEncoding fileID encoding for the table.
+   * @param instantTime instant time of interest.
+   * @param storage instance of {@link HoodieStorage}.
+   * @return Iterator of {@link HoodieRecord}s for RLI Metadata partition.
+   * @throws IOException
+   */
+  public static Iterator<HoodieRecord> generateRLIMetadataHoodieRecordsForBaseFile(String basePath,
+                                                                                   HoodieWriteStat writeStat,
+                                                                                   Integer writesFileIdEncoding,
+                                                                                   String instantTime,
+                                                                                   HoodieStorage storage) throws IOException {
+    String partition = writeStat.getPartitionPath();
+    String latestFileName = FSUtils.getFileNameFromPath(writeStat.getPath());
+    String previousFileName = writeStat.getPrevBaseFile();
+    String fileId = FSUtils.getFileId(latestFileName);
+    Set<String> recordKeysFromLatestBaseFile = getRecordKeysFromBaseFile(storage, basePath, partition, latestFileName);
+    if (previousFileName == null) {
+      return recordKeysFromLatestBaseFile.stream().map(recordKey -> (HoodieRecord)HoodieMetadataPayload.createRecordIndexUpdate(recordKey, partition, fileId,
+          instantTime, writesFileIdEncoding)).collect(toList()).iterator();
+    } else {
+      // read from previous base file and find difference to also generate delete records.
+      // we will return new inserts and deletes from this code block
+      Set<String> recordKeysFromPreviousBaseFile = getRecordKeysFromBaseFile(storage, basePath, partition, previousFileName);
+      List<HoodieRecord> toReturn = recordKeysFromPreviousBaseFile.stream()
+          .filter(recordKey -> {
+            // deleted record
+            return !recordKeysFromLatestBaseFile.contains(recordKey);
+          }).map(recordKey -> HoodieMetadataPayload.createRecordIndexDelete(recordKey)).collect(toList());
+
+      toReturn.addAll(recordKeysFromLatestBaseFile.stream()
+          .filter(recordKey -> {
+            // new inserts
+            return !recordKeysFromPreviousBaseFile.contains(recordKey);
+          }).map(recordKey ->
+              HoodieMetadataPayload.createRecordIndexUpdate(recordKey, partition, fileId,
+                  instantTime, writesFileIdEncoding)).collect(toList()));
+      return toReturn.iterator();
+    }
+  }
+
+  public static List<String> getRecordKeysDeletedOrUpdated(String basePath,
```

Comment:
method with same name in HoodieTableMetadataUtil  - getRecordKeysDeletedOrUpdated

Reply:
yes, this one is just for one HoodieWriteStat. the one in HoodieTableMetadataUtil is for an entire HoodieCommitMetadata